### PR TITLE
scikit-learn package

### PR DESCRIPTION
**Pull request by amueller**
This issue is mostly to track progress of the scikit-learn wrappers, but any feedback on me doing something stupid is obviously welcome.

Todo:

* [x] single classifier
* [x] classifier evaluator
* [x] add simple evaluation metric (``accuracy_score``)
* [x] add ``train_test_split``
* [x] add roc curve plotting
* [x] general scoring interface ~~add ``auc_roc_score``~~
* [x] cross_val_score
* [x] GridSearchCV
* [x] single transformer
* [x] Pipeline
* [x] single regressor
* [x] single clustering
* [x] auto-discover classifiers
* [x] auto-discover regressors
* [x] Add general scorers ~~auto-discover metrics~~
* [x] auto-discover feature selection
* [x] example: visualize GridSearchCV results
* [x] example: visualize decision function
* [x] example: PCA and manifold learning
* [ ] example: use dataset from another example
* ~~optional evaluation of decision_function vs caching.~~
* ~~pipeline with variable length many estimators~~ (not in this pr)
* [x] fix order of ports
* [x] testing
* [ ] user guide

### CODE REVIEW

**Comment by amueller** [comment]
I just pushed some manifold learning modules. Unfortunately they have a slightly different interface, as they don't have a ``transform`` or ``predict`` function, so they just server as a simple module that doesn't create a model.

**Comment by amueller** [comment]
I added some tests. Feedback welcome. They are not exhaustive but should make sure that the most common use cases work.

**Comment by amueller** [comment]
Oh and I am not sure if the test are currently run automatically as part of the test suite. where do I have to add them?

**Comment by remram44** [comment]
The tests will be run automatically if they can be imported [[see here]](https://travis-ci.org/VisTrails/VisTrails/jobs/46795758#L4479). Could you try adding `scikit-learn` to `requirements.txt`, `setup.py` or `.travis.yml`?

**Comment by amueller** [comment]
sweet, you enabled travis :)

**Comment by amueller** [comment]
I think the tests are running now. I still get ``WARNING: module has no tests: vistrails.packages.sklearn
`` but I guess that is because the tests are in ``vistrails.packages.sklearn.tests``. There is no message about ``vistrails.packages.sklearn.tests`` having no tests, I take that as a good sign.

**Comment by remram44** [comment]
Yes indeed. Great!

**Comment by amueller** [comment]
Sweet. I'm writing some user guide now.

I'm not sure where to put it, though.
Should it go under ``Intermediate Concepts and VisTrails Packages`` next to ``Example: Web Services`` as ``Example: Machine Learning``? Or just as ``Scikit-learn``? The two packages that I see documented are Web Services and ITK and they are both listed as "Example". And I'm not sure why ITK is listed under "Developer's Guilde".

**Comment by remram44** [comment]
It should go under `Intermediate Concepts and VisTrails Packages`.

Packages' docs are currently more of a collection of use-cases and examples than reference documentation.

**Comment by amueller** [comment]
Well I wouldn't call my short write-up a reference documentation.
But I think it is enough so that someone else can start using it.
From my side I think this is in a state that is mergable, let me know if you have any comments or concerns (I'll be back next Monday ;).

**Comment by remram44** [comment]
`test_classifier_training_predict` seems to fail randomly, see [c3a012a](https://travis-ci.org/VisTrails/VisTrails/jobs/46804259#L5172) and [225401a](https://travis-ci.org/VisTrails/VisTrails/jobs/46805962#L5172) (threshold is too high?)

**Comment by amueller** [comment]
yeah, I should just lower that.
I thought about fixing random seeds, but I think I prefer more robust tests.

**Comment by tacaswell** [comment]
Tests should almost always be done with fixed seeds.

**Comment by amueller** [comment]
@remram44 Monday seems to be off, are you there on Tuesday? Then I'd come over and we can discuss what is needed for merge?

**Comment by remram44** [comment]
Sure, I'll be there. Or you can just call in to our Wednesday meeting (you should have the info from the mailing-list).

**Comment by amueller** [comment]
Ok maybe I'll just do that then.

**Comment by amueller** [comment]
sweet, thanks.
Will it show up on the docs website or only after the next release?

**Comment by remram44** [comment]
It will show up on [the website](http://www.vistrails.org/usersguide/dev/html/) tomorrow, a cron on the webserver gets it nightly from the buildbot.
